### PR TITLE
Key generation depends only on the hash function.

### PR DESCRIPTION
**Pull request by armfazh**
changes:
 - Key generation depends only on the hash function.
 - formatting files
 - rename SUITES by RSABSSA, which is the actual name of the scheme.

### CODE REVIEW

File: examples/index.ts [hunk 1]
```diff
@@ -3,18 +3,19 @@
 
 import { webcrypto } from 'node:crypto';
 
+import { RSABSSA } from '../src/index.js';
+
```
Comment (thibmeu) [inline examples/index.ts:7]: nits:
```suggestion
```

File: src/blindrsa.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) 2023 Cloudflare, Inc.
 // Licensed under the Apache-2.0 license found in the LICENSE file or at https://opensource.org/licenses/Apache-2.0
 
+import sjcl from './sjcl/index.js';
+
```
Comment (thibmeu) [inline src/blindrsa.ts:5]: nits:
```suggestion
```

File: src/blindrsa.ts [hunk 3]
```diff
@@ -13,40 +15,41 @@ import {
...
-
 export enum PrepareType {
     Deterministic = 0,
     Randomized = 32,
 }
 
 export type BlindOutput = { blindedMsg: Uint8Array; inv: Uint8Array };
 
+export interface BlindRSAParams {
+    name: string;
```
Comment (thibmeu) [inline src/blindrsa.ts:26]: `name` does not appear to be used. Is it meant for debugging? In this case, either mark it as optional or remove it

File: src/blindrsa.ts [hunk 4]
```diff
@@ -210,29 +213,32 @@ export class BlindRSA {
...
+        const algorithm = { name: BlindRSA.NAME, saltLength: this.params.saltLength };
         if (!(await crypto.subtle.verify(algorithm, publicKey, sig, msg))) {
             throw new Error('invalid signature');
         }
 
         return sig;
     }
 
+    static generateKey(
+        algorithm: Pick<RsaHashedKeyGenParams, 'modulusLength' | 'publicExponent' | 'hash'>,
```
Comment (thibmeu) [inline src/blindrsa.ts:225]: good use of `Pick`

File: src/blindrsa.ts [hunk 5]
```diff
@@ -210,29 +213,32 @@ export class BlindRSA {
...
+    ): Promise<CryptoKeyPair> {
+        return crypto.subtle.generateKey({ ...algorithm, name: 'RSA-PSS' }, true, [
```
Comment (thibmeu) [inline src/blindrsa.ts:227]:
```suggestion
        return crypto.subtle.generateKey({ ...algorithm, name: BlindRSA.NAME }, true, [
```

File: src/index.ts [hunk 6]
```diff
@@ -3,38 +3,59 @@
...
 // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-rsa-blind-signatures-14
-import { BlindRSA, PrepareType } from './blindrsa.js';
+import { BlindRSA, PrepareType, type BlindRSAParams } from './blindrsa.js';
 
-export type { BlindRSA };
+export { BlindRSA, type BlindRSAParams };
 
-export const SUITES = {
+export const Params: Record<string, BlindRSAParams> = {
+    RSABSSA_SHA384_PSS_Randomized: {
```
Comment (thibmeu) [inline src/index.ts:11]: you may want to document the use of `RSABSSA`. you mention `RSABSSA, which is the actual name of the scheme`, which might point to a reference

**Review by thibmeu — APPROVED** [review]
only minor comments. good to go

File: src/blindrsa.ts (hunk 3 above)
Comment (armfazh) [reply src/blindrsa.ts:26]: there is a top-level function `getSuiteByName` which uses the `name` field.

File: src/index.ts (hunk 6 above)
Comment (armfazh) [reply src/index.ts:11]: fun fact: rfc9474 was just published today.
so all documentation links need to be updated.

Comment (thibmeu) [reply src/index.ts:11]: good news and more work

Comment (thibmeu) [reply src/index.ts:11]: [my understanding of the publication state is it got assigned a number but needs approvals for all the authors]

**Pull request merged**